### PR TITLE
Create unique tempdir to avoid name collisions

### PR DESCRIPTION
**Pull request by MarcoFalke**
Otherwise, I believe this could lead to collisions when the bench is run more than once per day.

### CODE REVIEW

File: runner/run_bench.py [hunk 1]
```diff
@@ -97,13 +96,15 @@ def run_benches():
     Create a tmp directory in which we will clone bitcoin, build it, and run
     various benchmarks.
     """
-    workdir: Path = _create_working_dir()
+    workdir = Path(tempfile.mkdtemp(prefix=
+        f"bench-{REPO_BRANCH}-"
+        f"{datetime.datetime.utcnow().strftime('%Y-%m-%d')}-"))
+    WORKING_DIR_NAME = workdir
```
Comment (jamesob) [inline runner/run_bench.py]: Unfortunately this won't work as-written (and is why I have the clunky `RUN_DATA` container object above, which I think you should use):

```python
In [1]: WORK_DIR = None

In [2]: def set_work():
   ...:     WORK_DIR = "yo"
   ...:

In [3]: set_work()

In [4]: WORK_DIR

In [5]:
```

You need to use the `global` keyword in order to replace references on scope-inherited symbols.

Comment (MarcoFalke) [reply runner/run_bench.py]: uups. Good catch

**Comment by MarcoFalke** [comment]
@jamesob Is that better?

**Comment by MarcoFalke** [comment]
Ah, going to use RUN_DATA ...

**Comment by MarcoFalke** [comment]
Rebased

**Comment by jamesob** [comment]
Awesome, thanks

**Pull request merged**